src/t8_forest/t8_forest_adapt.cxx

```diff
@@ -281,23 +284,34 @@ t8_forest_adapt (t8_forest_t forest)
         is_family = 0;
 #endif
       }
-      T8_ASSERT (!is_family || tscheme->t8_element_is_family (elements_from));
+      T8_ASSERT (!is_family || tscheme->t8_element_is_family (elements_from));     
+      
       /* Pass the element, or the family to the adapt callback.
-       * The output will be > 0 if the element should be refined
-       *                    = 0 if the element should remain as is
-       *                    < 0 if we passed a family and it should get coarsened.
-       */
-      refine =
-        forest->set_adapt_fn (forest, forest->set_from, ltree_id,
-                              el_considered, tscheme, num_elements,
-                              elements_from);
+       * The output will be 
+       *
+       *      < 0 if we passed a family and it should get coarsened 
+       *      = 0 if the element should remain as is 
+       *      = 1 if the element should be refined (using the chosen recursive refinement scheme)
+       *      > 1 if we use subelements
+       * 
+       * The values -1,0 and 1 will appear, if we use the standard refinement scheme of the given eclass.
+       * 
+       * Values above 1 will correspond to specific subelement types. The values from 2 to 65 are reserved for subelement types that can remove hanging nodes from a given scheme.
+       *  
+       * For example the refine values for the 2D Quad scheme will be between -1 and 16. The values -1, 0 and 1 are for the standard refinement
+       * and the values 2 to 16 correspond to the subelement types 1 to 15 (0001 to 1111 in base 2) and will be used by the element files of the quad 
+       * scheme in order to remove hanging nodes */ 
+
+      refine = forest->set_adapt_fn (forest, forest->set_from, ltree_id,
+                                     el_considered, tscheme, num_elements,
+                                     elements_from);
+      
       T8_ASSERT (is_family || refine >= 0);
-      if (refine > 0 && tscheme->t8_element_level (elements_from[0]) >=
-          forest->maxlevel) {
+      if (refine > 0 && tscheme->t8_element_level (elements_from[0]) >= forest->maxlevel) {
         /* Only refine an element if it does not exceed the maximum level */
         refine = 0;
```

Comment:
Indeed, there might be a problem which is to solve later. In order to remove hanging nodes it should be possible to refine elements of the highest level by using subelements. But thinking of numerical accuracy it should not be possible to refine an element of the highest level anymore (neighter by a standard refinement, nor by subelements).

Reply:
Note that for the hanging node case we would only need subelements of maxlevel-1.
Since we only build subelements if a neighbor element is refined deeper, which cannot happen if we have maxlevel ourselves...

Reply:
Ah yes, I see :D 

So in case of removing hanging nodes, this shouldnt be a problem since we would never refine an element of the highest level. 

But also in general I wouldnt allow a max level element to be refined using subelements. Even if the level of the subelement stays the same intern, the coordinates would change to values that might be problematic from a numerical point of view (I see no difference there between standard refinement and subelements).

Reply:
Okay, so we do not allow subelements for max level elements. I am fine with that.